src/HasApiTokens.php

```diff
@@ -70,22 +82,18 @@ public function createToken($name, array $scopes = [])
 
     /**
      * Get the user provider name.
-     *
-     * @return string|null
      */
-    public function getProvider(): ?string
+    public function getProvider(): string
     {
         $providers = collect(config('auth.guards'))->where('driver', 'passport')->pluck('provider')->all();
 
         foreach (config('auth.providers') as $provider => $config) {
-            if (in_array($provider, $providers)
-                && (($config['driver'] === 'eloquent' && is_a($this, $config['model']))
-                    || ($config['driver'] === 'database' && $config['table'] === $this->getTable()))) {
+            if (in_array($provider, $providers) && $config['driver'] === 'eloquent' && is_a($this, $config['model'])) {
                 return $provider;
             }
         }
 
-        return null;
+        throw new LogicException('Unable to determine authentication provider for this model from configuration.');
```

Comment:
So what should they do in this case?

Reply:
This should never happen, the authenticatable model should have a provider registered on `auth` configuration as we instruct the developer when [installing Passport](https://laravel.com/docs/11.x/passport#installation:~:text=Finally%2C%20in%20your,incoming%20API%20requests%3A)